src/api/fetch.js

```diff
@@ -0,0 +1,18 @@
+import baseURL from "./base_url";
+
+export function fetchIndex(url, page, auth) {
```

Comment:
Just `index` as a method name is fine. The fact that it is in the `api` module makes clear that it will do fetching. Also, rename `url` to `path` since it's not a URL, it's a path.